### PR TITLE
Added : Login  Functionality and Design.

### PR DESCRIPTION
**Pull request by shivamsherje**
## Discription : 
-  Added login functionality.
-  Added Design of login functionality.

# Screenshots : 
Screenshot 2023-10-26 at 3 31 42 PM

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -3,7 +3,16 @@ import { createRouter, createWebHistory } from 'vue-router'
 const router = createRouter({
   history: createWebHistory(import.meta.env.BASE_URL),
   routes: [
-   
+    {
+      path: '/',
+      name: 'home',
+      component: () => import('../views/users/TheLogin.vue')
```
Comment (AishwaryShrivastav) [inline src/router/index.ts]: Please remove this

File: src/stores/auth.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            username,
+            password,
+            grant_type: 'password',
+            client_id: import.meta.env.VITE_CLIENT_ID
+          }
+        })
+        if (response.status === 200) {
+          this.isLoggedIn = true
+          this.authToken = response.data.access_token
+          console.log('Login Successful !!')
```
Comment (AishwaryShrivastav) [inline src/stores/auth.ts]: remove console logs

File: src/stores/auth.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        } else {
+          console.log('Error in Login !!')
```
Comment (AishwaryShrivastav) [inline src/stores/auth.ts]: console

File: src/main.ts [hunk 4]
```diff
@@ -1,14 +1,22 @@
+import 'vuetify/styles'
+import { createVuetify } from 'vuetify'
+import * as components from 'vuetify/components'
+import * as directives from 'vuetify/directives'
+import '@mdi/font/css/materialdesignicons.css'
 
+const vuetify = createVuetify({
```
Comment (AishwaryShrivastav) [inline src/main.ts:14]: move this after imports

File: src/views/users/TheLogin.vue [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import { useAuthStore } from '../../stores/auth'
+
+const auth = useAuthStore()
+
+const inputData = ref({
+  username: '',
+  password: ''
+})
+
+const show2 = ref(false)
```
Comment (AishwaryShrivastav) [inline src/views/users/TheLogin.vue]: pls check this

File: src/views/users/TheLogin.vue [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+<template>
+  <v-dialog v-model="dialog" width="450" transition="dialog-top-transition">
+    <v-card class="pa-9 rounded-xl" id="login-modal-card">
+      <p class="font-weight-black mb-6 mt-6">Open Llama</p>
+      <v-form @submit.prevent="submitLoginForm">
+        <p class="font-weight-light">Please enter your Id/Email</p>
+        <v-text-field class="login-input-fields" clearable type="text" density="compact" placeholder="id or email"
+          variant="outlined" v-model.trim="inputData.username" :rules="usernameRules" rounded="lg"></v-text-field>
+        <p class="font-weight-light">Please enter your Password</p>
```
Comment (AishwaryShrivastav) [inline src/views/users/TheLogin.vue]: add logical  spacing

File: src/views/users/TheLogin.vue [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+          type="submit">
+          continue
+        </v-btn>
+      </v-form>
+    </v-card>
+  </v-dialog>
+</template>
+
+<style scoped>
+@import url('https://fonts.googleapis.com/css2?family=Albert+Sans&family=Dosis:wght@700&family=Plus+Jakarta+Sans&display=swap');
```
Comment (AishwaryShrivastav) [inline src/views/users/TheLogin.vue]: font import will go to app.vue

File: package.json [hunk 8]
```diff
@@ -15,9 +15,15 @@
     "format": "prettier --write src/"
   },
   "dependencies": {
+    "@mdi/font": "^7.3.67",
```
Comment (disha1202) [inline package.json:18]: this can be removed as we'll be using iconsax

File: package.json [hunk 9]
```diff
@@ -15,9 +15,15 @@
...
+    "axios": "^1.5.1",
+    "dotenv": "^16.3.1",
```
Comment (disha1202) [inline package.json]: Remove this

File: package.json [hunk 10]
```diff
@@ -15,9 +15,15 @@
...
     "pinia": "^2.1.7",
     "vue": "^3.3.4",
-    "vue-router": "^4.2.5"
+    "vue-i18n": "^9.6.0",
+    "vue-router": "^4.2.5",
+    "vuesax": "^3.12.2",
```
Comment (disha1202) [inline package.json]: Remove this

File: src/stores/auth.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      } catch (error) {
+        console.error(error)
+        return {}
+      }
+    },
+    logout() {
+      this.isLoggedIn = false
+      this.authToken = null
+    }
+  }
```
Comment (disha1202) [inline src/stores/auth.ts]: Use composition API

File: src/views/users/TheLogin.vue [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+const auth = useAuthStore()
+
+const inputData = ref({
+  username: '',
+  password: ''
+})
+
+const showPassword = ref(false)
+const dialog = ref(false)
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: Improve variable name

File: src/views/users/TheLogin.vue [hunk 13]
```diff
@@ -0,0 +1,144 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue'
+import { useAuthStore } from '../../stores/auth'
+
+const auth = useAuthStore()
+
+const inputData = ref({
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: improve variable name, also why is there a need for input data object?

File: src/views/users/TheLogin.vue (hunk 12 above)
Comment (shivamsherje) [inline src/views/users/TheLogin.vue]:
```suggestion
const loginModal = ref(false)
```

File: src/views/users/TheLogin.vue [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+})
+
+const showPassword = ref(false)
+const dialog = ref(false)
+
+const submitLoginForm = async () => {
+  await auth.login(inputData.value.username, inputData.value.password)
+}
+
+const usernameRules = ref([(v: any) => !!v || 'Username is required'])
```
Comment (shivamsherje) [inline src/views/users/TheLogin.vue]:
```suggestion
const usernameValidation = ref([(v: any) => !!v || 'Username is required'])
```

File: src/views/users/TheLogin.vue [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+const passwordRules = ref([
```
Comment (shivamsherje) [inline src/views/users/TheLogin.vue]:
```suggestion
const passwordValidation = ref([
```

File: src/views/users/TheLogin.vue [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+const inputData = ref({
+  username: '',
+  password: ''
+})
+
+const showPassword = ref(false)
+const dialog = ref(false)
+
+const submitLoginForm = async () => {
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: Improve method name

File: src/views/users/TheLogin.vue (hunk 15 above)
Comment (disha1202) [inline src/views/users/TheLogin.vue]: Improve variable names

File: src/views/users/TheLogin.vue [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+</script>
+
+<template>
+  <v-dialog v-model="dialog" width="450" transition="dialog-top-transition">
+    <v-card class="pa-9 rounded-xl" id="login-modal-card">
+      <p class="font-weight-black mb-6 mt-6">Open Llama</p>
+
+      <v-form @submit.prevent="submitLoginForm">
+        <vs-icon icon="mood"></vs-icon>
+        <p class="font-weight-light">Please enter your Id/Email</p>
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: internationalize static text

File: src/views/users/TheLogin.vue [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+})
+</script>
+
+<template>
+  <v-dialog v-model="dialog" width="450" transition="dialog-top-transition">
+    <v-card class="pa-9 rounded-xl" id="login-modal-card">
+      <p class="font-weight-black mb-6 mt-6">Open Llama</p>
+
+      <v-form @submit.prevent="submitLoginForm">
+        <vs-icon icon="mood"></vs-icon>
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: Remove this

File: src/views/users/TheLogin.vue (hunk 17 above)
Comment (disha1202) [inline src/views/users/TheLogin.vue]: We only have support for username

File: src/views/users/TheLogin.vue [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+          variant="outlined"
+          :type="showPassword ? 'text' : 'password'"
+          @click:append-inner="showPassword = !showPassword"
+          density="compact"
+          placeholder="password"
+          rounded="lg"
+        >
+          <template v-slot:append-inner>
+            <i class="isax isax-eye" v-if="showPassword" @click="showPassword = !showPassword"></i>
+            <i class="isax isax-eye-slash" v-else @click="showPassword = !showPassword"></i>
```
Comment (disha1202) [inline src/views/users/TheLogin.vue:61]:
```suggestion
            <i :class="showPassword ? 'isax isax-eye' : 'isax isax-eye-slash'" @click="showPassword = !showPassword"></i>
```

File: src/views/users/TheLogin.vue [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+          </template>
+        </v-text-field>
+
+        <div class="login-helpers">
+          <v-checkbox label="Remember me" color="orange" hide-details></v-checkbox>
+          <p class="forgot-password-button">Forgot Password?</p>
+        </div>
+        
+        <v-btn
+          color="#FB8430"
```
Comment (disha1202) [inline src/views/users/TheLogin.vue:71]: Make this global css variable

File: src/views/users/TheLogin.vue [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+
+<style scoped>
+#login-details-submit-button {
+  display: block;
+  margin: 0 auto;
+  width: 60%;
+  height: 50px;
+  font-weight: 600;
+  font-family: 'Albert Sans', sans-serif;
+  color: #fff;
```
Comment (disha1202) [inline src/views/users/TheLogin.vue]: Why do we need extra css. Try to use Vuetify classes.

File: src/theme/variable.css [hunk 22]
```diff
@@ -0,0 +1,8 @@
+@import url('https://fonts.googleapis.com/css2?family=Albert+Sans&family=Dosis:wght@700&family=Plus+Jakarta+Sans&display=swap');
```
Comment (AishwaryShrivastav) [inline src/theme/variable.css:1]: is this necessary here ? @disha1202 please check

File: src/views/users/TheLogin.vue (hunk 17 above)
Comment (AishwaryShrivastav) [inline src/views/users/TheLogin.vue]: Username instead of Email

File: tsconfig.app.json [hunk 23]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "@vue/tsconfig/tsconfig.dom.json",
-  "include": ["env.d.ts", "src/**/*", "src/**/*.vue"],
+  "include": ["env.d.ts", "src/**/*", "src/**/*.vue", "package.json"],
```
Comment (disha1202) [inline tsconfig.app.json]: Why is this needed?

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/stores/auth.ts [hunk 24]
```diff
@@ -0,0 +1,37 @@
+import { defineStore } from 'pinia'
+import axios from 'axios'
+
+export const useAuthStore = defineStore({
+  id: 'auth',
+  state: () => ({
+    isLoggedIn: false,
```
Comment (disha1202) [inline src/stores/auth.ts]: Why do we need an extra bool variable to verify whether the user is logged in or not?

**Review by disha1202 — CHANGES_REQUESTED** [review]
